torchvision/datasets/imagenet.py

```diff
@@ -118,54 +108,120 @@ def extra_repr(self):
         return "Split: {split}".format(**self.__dict__)
 
 
-def parse_devkit(root):
-    idx_to_wnid, wnid_to_classes = parse_meta(root)
-    val_idcs = parse_val_groundtruth(root)
-    val_wnids = [idx_to_wnid[idx] for idx in val_idcs]
-    return wnid_to_classes, val_wnids
+def _verify_archive(archive, md5, force):
```

Comment:
I'm a little unhappy with having `_verify_archive` as function and `ImageNet._verify_archive()` as method with similar signatures. Due to different error messages I haven't found a good way to unify them. Do you see a way or do we keep it as is for now?

Reply:
Do we need to have the method one? Can't we just change the error message in the function? I'm not entirely clear on what `force` flag is meant to do: if we `md5` doesn't match, we should just raise an error anyway, as the file is corrupted. In which situation we would not want to raise an error?

Reply:
> Do we need to have the method one? Can't we just change the error message in the function?

I was thinking since `parse_*_archive()` is called without a root directory the error messages should differ. Do you want me to change the error message of the `_verify_archive` function to the root / file scheme?

> I'm not entirely clear on what force flag is meant to do: if we md5 doesn't match, we should just raise an error anyway, as the file is corrupted. In which situation we would not want to raise an error?

I was thinking what if another dataset, for example an drop-in replacement, uses the same structure as the `ImageNet` dataset? In that case we couldn't use the `parse_*_archive()` functions, since the MD5 checksums wouldn't match. Thus, I've implemented the `force` flag which allows the parsing if the user explicitly wants to. Since it defaults to `False`, parsing the archive with the dataset object never forces an extraction.

Reply:
> Do you want me to change the error message of the _verify_archive function to the root / file scheme?

This sounds simpler to me than having a duplicate function.

> I was thinking what if another dataset, for example an drop-in replacement, uses the same structure as the ImageNet dataset?

Is this for testing?
I don't think we would need such a functionality for now. The dataset is very similar to ImageFolder, so I would expect most users would just use it instead. And if users start requesting this functionality to `ImageNet` we can look into adding it, but for now I think it's not needed.